modules/ldap/src/main/java/com/neighborhoods/wasabi/ldap/impl/DirectoryAuthorization.java

```diff
@@ -0,0 +1,406 @@
+    }
+
+    /**
+     * Set a user to super admin role
+     * 
+     * @throws UnsupportedOperationException for all calls to this method
+     */
+    @Override
+    public void assignUserToSuperAdminRole(final UserInfo candidateUserInfo, final UserInfo assigningUserInfo) {
+        throw new UnsupportedOperationException(DIRECTORY_OPERATION_NOT_SUPPORTED);
+    }
+
+    /**
+     * Validate the given user is a super admin. Note that unauthorized access attempts will cause the user to be force
+     * logged out.
+     * 
+     * @param userID the target user
+     * @throws AuthenticationException when the user is not a superadmin
+     */
+    @Override
+    public void checkSuperAdmin(UserInfo.Username userID) {
+        DirectoryUser user = this.userDirectory.lookupDirectoryUser(userID.getUsername());
+        if (!Role.toRole(user.getRole()).equals(Role.SUPERADMIN)) {
+            throw new AuthenticationException("error, user " + userID + " is not a superadmin");
+        }
+    }
+
+    /**
+     * Validate user has permission for the given application
+     * 
+     * @param userID the user ID
+     * @param applicationName the target application
+     * @throws AuthenticationException for unauthorized access
+     */
+    @Override
+    public void checkUserPermissions(UserInfo.Username userID, Application.Name applicationName,
+            Permission permission) {
+        // get the user's permissions for this applicationName
+        UserPermissions userPermissions = getUserPermissions(userID, applicationName);
+        // check that the user is permitted to perform the action
+        if (userPermissions == null || !userPermissions.getPermissions().contains(permission)) {
+            throw new AuthenticationException("error, user " + userID + " not authorized to " + permission.toString()
+                    + " on application " + applicationName.toString());
+        }
+    }
+
+    /**
+     * Delete a user role
+     * 
+     * @throws UnsupportedOperationException for all calls to this method
+     */
+    @Override
+    public void deleteUserRole(UserInfo.Username userID, Application.Name applicationName, UserInfo admin) {
+        throw new UnsupportedOperationException(DIRECTORY_OPERATION_NOT_SUPPORTED);
+    }
+
+    /**
+     * Gets the all application name from application list.
+     *
+     * @return list of all application names
+     */
+    List<String> getAllApplicationNamesFromApplicationList() {
+        Result<ApplicationList> allAppNames = applicationListAccessor.getUniqueAppName();
+        return StreamSupport
+                .stream(Spliterators.spliteratorUnknownSize(allAppNames.iterator(), Spliterator.ORDERED), false)
+                .map(t -> t.getAppName()).collect(Collectors.toList());
+    }
+
+    /**
+     * Retrieve all users of the application. Note that all application based permissions has been collapsed.
+     * 
+     * @param the application name
```

Comment:
Should be "@param applicationName the application name" to keep the IDE from complaining.